src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1912,13 +1912,24 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     // - This is related to work done for GH#2988.
     void ControlCore::GotFocus()
     {
-        _terminal->FocusChanged(true);
+        _focusChanged(true);
     }
 
     // See GotFocus.
     void ControlCore::LostFocus()
     {
-        _terminal->FocusChanged(false);
+        _focusChanged(false);
+    }
+
+    void ControlCore::_focusChanged(bool focused)
+    {
+        // GH#13461 - temporarily turn off read-only mode, send the focus event,
+        // then turn it back on. Even in focus mode, focus events are fine to
+        // send. We don't want to pop a warning every time the control is
+        // focused.
+        const auto previous = std::exchange(_isReadOnly, false);
+        const auto restore = wil::scope_exit([&]() { _isReadOnly = previous; });

```

Comment:
nit: we don't seem to use `restore` anywhere, do we need it?

Reply:
yea, the way `wil::scope_exit` works, you _have_ to stash the result. When `restore` goes out of scope, then the body of the `scope_exit` will fire. If you don't stash it in a local, it'll destruct itself immediately, which in this case would automatically restore `_isReadOnly`

_(this is a mistake I've made in the past)_

Reply:
Ah gotcha, thanks!